src/bedrock/api.js

```diff
@@ -94,8 +94,113 @@ module.exports = class BedrockRealmAPI extends RealmAPI {
     return await this.rest.put(`/worlds/${realmId}/${state}`)
   }
 
+  async deleteRealm (realmId) {
+    return await this.rest.delete(`/worlds/${realmId}`)
+  }
+
   async getRealmWorldDownload (realmId, slotId, backupId = 'latest') {
     const data = await this.rest.get(`/archive/download/world/${realmId}/${slotId}/${backupId}`) // if backupId = latest will get the world as it is now not the most recent backup
     return new Download(this, data)
   }
+
+  async getRealmSubscriptionInfo (realmId) {
+    return await this.rest.get(`/subscriptions/${realmId}`)
+  }
+
+  async getRealmSubscriptionInfoDetailed (realmId) {
+    return await this.rest.get(`/subscriptions/${realmId}/details`)
+  }
+
+  async changeRealmActiveSlot (realmId, slotId) {
+    return await this.rest.put(`/worlds/${realmId}/slot/${slotId}`)
+  }
+
+  async changeRealmNameAndDescription (realmId, name, description) {
+    return await this.rest.put(`/worlds/${realmId}`, {
+      body: {
+        name,
+        description
+      }
+    })
+  }
+
+  async resetRealm (realmId) {
+    return await this.rest.put(`/worlds/${realmId}/reset`)
+  }
+
+  // Reference https://github.com/PrismarineJS/prismarine-realms/issues/34 for configuration structure
+  async changeRealmConfiguration (realmId, configuration) {
+    return await this.rest.put(`/worlds/${realmId}/configuration`, {
+      body: configuration
+    })
+  }
+
+  async removeRealmInvite (realmId, uuid) {
```

Comment:
Maybe `removePlayerFromRealm` may be a better name here to draw a clear distinction from `rejectRealmInvite`.